flink-filesystems/flink-s3-fs-hadoop/src/main/java/org/apache/hadoop/fs/s3a/S3AInstrumentation.java

```diff
@@ -0,0 +1,1670 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.s3a;
```

Comment:
this seems to be a new class that is not called. Also this does not have any unit tests.
This seems to be a fixup in Hadoop - is this the appropriate repo to make this fix?
Is there an upstream repo to fix this and then for us to pick up the change.

Reply:
Yes, it just overwrite the S3AInstrumentation class of hadoop-aws, and the memory use I think is bad usage of flink, not to close FileSystem or unregister S3A metrics, the arch of flink not to close filesystem but only the output stream;
So the upper hadoop common seems not need to correct, there are two simple chooses for once, one is degress hadoop-aws to prior 2.8, or overwrite S3A.